File: study/2022/22.11.15/EJH/1475_방 번호.py
import math

N = int(input())

arr = [0 for i in range(9)]

while N > 0:
    num = N % 10
    if num == 9:
        num = 6
    arr[num] += 1

    N //= 10

arr[6] = math.ceil(arr[6] / 2.0)

print(max(arr))

File: study/2022/22.11.15/HJH/1475. 방 번호.java
import java.io.*;

public class Main {
    public static void main(String[] args) throws Exception {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int N = Integer.parseInt(br.readLine());
        int[] num = new int[10];
        int ans = 1;
        int max = 1;
        if (N < 10) System.out.print(ans);
        else if (N == 1000000) System.out.print(6);
        else {
            while (N > 0) {
                num[N % 10]++;
                N = N / 10;
            }
            num[6] = (num[6] + num[9] + 1) / 2;
            for (int i = 0; i <= 8; i++) max = Math.max(max, num[i]);
            System.out.print(max);
        }
        br.close();
    }
}
